src/Bicep.Core/SemanticModel/Compilation.cs

```diff
@@ -88,7 +108,13 @@ private SemanticModel GetSemanticModelInternal(SyntaxTree syntaxTree)
             // allow type queries now
             symbolContext.Unlock();
 
-            return new SemanticModel(file, symbolContext.TypeManager, bindings);
+            foreach (var targetScopeSyntax in syntaxTree.ProgramSyntax.Children.OfType<TargetScopeSyntax>())
+            {
+                // force a type check to get diagnostics for the target scope
```

Comment:
>force [](start = 19, length = 5)

Why?

Reply:
Determining the targetScope happens before name binding, so I use syntax rather than types. Which means that no one actually requests the type check during a compilation unless I explicitly force it.